proposals/manegiable-aff-data-mapping/proposal.md

```diff
@@ -0,0 +1,234 @@
+
+Variables (Runtime Data Holders): A mechanism to define reusable, structured **client-side** runtime data.
+
+Capabilities / Clusters / Or? (Context and Management): A new top-level container that defines a stateful runtime context. It groups related affordances and with a client state, establishing an explicit relationships between interactions (or group).
+
+By establishing these two mechanisms, the TD can effectively model a very simple state machine and explicitly link different part of low level messages to the application layer.
+
+---
+
+## 2. Scope & Goals
+
+* Fix the identified limitations in the WoT TD for complex interaction patterns.
+* Define a simple yet powerful mechanism to cover most of the analyzed user stories and **implementations**.
+
+---
+
+## 4. Proposed Core Mechanisms
+
+This section defines the two new, orthogonal concepts.
+
+### 4.1 Concept A: Variables (Client side Runtime Data Modeling)
+
+In the current TD model we focused on describing interactions patterns between a client and a Thing. However, complex interactions often require the client to manipulate and manage the data exchanged with the remote Thing. To address this, we introduce the concept of **Variables**. 
+
+* **Definition:** **Variables** are internal, logical data structures defined within the TD scope to hold **runtime data** essential for complex interactions. They are *not* automatically network-exposed Properties.
+* **Role:** Serve as a clean way to describe the **mapping of runtime data** in an otherwise static document.
+* **Declaration Machanism:** A variable can be decleared in any json schema that used in the TD (e.g., as part of an Action input or output schema). Variables are identified by a unique name.
+* **Usage:** A variable can be referenced in children elements of Affordances (e.g., in `forms` or `href` templates) using a defined syntax (e.g., `${variableName}`). When referenced, the client MUST create a two-way binding between the variable and the corresponding message component.
+
+Example supporting [Data Mapping in HTTP](https://github.com/w3c/wot-thing-description/blob/main/planning/work-items/analysis/analysis-data-mapping.md#user-stories):
+```json
+{
+  "actions": {
+    "startJob": {
+      "input": {
+        "type": "object",
+        "properties": {
+          "jobType": { "type": "string", "variable": "jobType" },
+          "priority": { "type": "integer" }
+        }
+      },
+      "forms": [
+        {
+          "href": "http://example.com/jobs",
+          "htv:methodName": "POST",
+          "http:headers": {
+            "X-Job-Type": "${jobType}",
+          }
+        }
+      ]
+    }
+  },
+}
+```
+In this example, the `jobType` property in the Action input schema is marked as a Variable. When the client invokes the `startJob` action, it will bind the value of `jobType` to the `X-Job-Type` HTTP header in the request and the real payload sent the the server will not contain the `jobType` field :
```

Comment:
```suggestion
In the example above, the `jobType` property in the Action input schema is marked as a Variable. When the client invokes the `startJob` action, it will bind the value of `jobType` to the `X-Job-Type` HTTP header in the request and the real payload sent to the the server will not contain the `jobType` field.
```